metricbeat/mb/module/wrapper.go

```diff
@@ -311,6 +302,27 @@ func (msw *metricSetWrapper) Test(d testing.Driver) {
 	})
 }
 
+func (msw *metricSetWrapper) handleFetchError(err error, reporter mb.PushReporterV2) {
+	if err != nil {
```

Comment:
what about "switching" to a switch statement, there are enough conditions here I think listing them all in one place would be helpful for understanding in the future.